### ISSUE TITLE
"Extends" with plugins displays a confusing error message

### ISSUE THREAD

**Issue by danny-andrews** [eslint/eslint#8192]
**Tell us about your environment**

* **ESLint Version:** 3.16.1
* **Node Version:** 7.6.0
* **yarn Version:** 0.20.3

**What parser (default, Babel-ESLint, etc.) are you using?** babel-eslint

**Please show your full configuration:**
```js
module.exports = {
  extends: ['airbnb', 'plugin:flowtype/recommended'],
  plugins: ['flowtype'],
  rules: {
    'flowtype/define-flow-type': 1,
    'flowtype/use-flow-type': 1,
    'flowtype/delimiter-dangle': [2, 'always'],
    'flowtype/semi': [2, 'always'],
  },
  settings: {
    flowtype: {
      onlyFilesWithFlowAnnotation: true,
    },
  },
};
```

package.json:
```json
...
"eslint-plugin-flowtype": "2.30.0",
...
```

**What did you expect to happen?**
My code would be linted with 'flowtype' rules.

**What actually happened? Please include the actual, raw output from ESLint.**
I get:
```
Oops! Something went wrong! :(

ESLint couldn't find the config "plugin:flowtype/recommended" to extend from. Please check that the name of the config is correct.
```

**Comment by danny-andrews** [comment]
Note: when I rollback to ~3.15.0, this works. It appears to be a regression in 3.16.x.

**Comment by platinumazure** [comment]
Does it work if you rollback 360dbe4? That's the only commit in the release notes that I can find that seems to relate to config extension. (PR: #8100, issue: #6115)

**Comment by danny-andrews** [comment]
Yep, that's the offending commit. Nice hunting!

**Comment by alberto** [comment]
@danny-andrews What do you mean by "it works"? 
The report is complaining about the config, when the actual error I get if I revert the commit is "Error: Cannot find module 'babel-eslint'"
Is that what you mean?

**Comment by danny-andrews** [comment]
@alberto Yes, I guess I thought that was a different error.
To clarify:
On 360dbe4:
  without `babel-eslint` installed: `ESLint couldn't find the config "plugin:flowtype/recommended" to extend from. Please check that the name of the config is correct.`
  with `babel-eslint` installed: Works as expected with no errors.
On Pre-360dbe4:
  without `babel-eslint` installed: `Cannot find module 'babel-eslint'`
  with `babel-eslint` installed: Works as expected with no errors.

**Comment by alberto** [comment]
Thanks @danny-andrews for confirming, that's what I was experiencing.
